modules/apps/foundation/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/servlet/FriendlyURLServlet.java

```diff
@@ -99,7 +99,10 @@ public Redirect getRedirect(HttpServletRequest request, String path)
 		Group group = groupLocalService.fetchFriendlyURLGroup(
 			companyId, friendlyURL);
 
-		if (group == null) {
+		if ((group == null) ||
+			(group.isRegularSite() &&
+			 Validator.isNumber(friendlyURL.substring(1)))) {
+
 			String screenName = friendlyURL.substring(1);
```

Comment:
I think this check is not necessary. If a site has a numeric friendlyURL before, line 99 will return the site. Plus, we won't be able to create a user with a screen name that matches this numeric friendlyURL (groupId) since it will throw a duplicate friendlyURL exception. 

Speaking of which, it makes me think that we should add an upgrade process to replace all site's numeric friendlyURLs with a prefix, same as what we do for making the default url when user empties the friendlyURL (https://github.com/pei-jung/liferay-portal/pull/207/commits/0e4d2cf9648718df6377f1135058614c60533ace). Also note that the default here is different from what is done on initial creation because the site name can be empty if the inputted site name is not in Latin character. We have to make sure we provide a default url that will work all the time.

Any thoughts @jorgeferrer?

Reply:
Good catch, @pei-jung.  This code is completely unnecessary :)